frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -319,6 +319,7 @@ function DataFrame({
       }
     }),
     [
+      columns,
```

Comment:
**suggestion:** This would have been caught by our tooling if it didn't have disabled eslint rules. Can we refactor this so that the callback definition adheres to the rules of hooks?

Reply:
Sounds good 👍  I split it up in a way that allows full dependency inspection. However, the only way I found that fully allows this without breaking the debounce functionality is by splitting it into two callbacks and a memo. I will likely move this refactor into a dedicated PR since it might need some more thoughts and iteration and will do it for syncSelectionState + syncEditingState.

Reply:
Yeah sounds good. I think a good outcome of that refactor would be if we could have a shared `useDebouncedCallback` hook that handles this for us since I'm sure it'll be useful in many other places outside of this file too!

Reply:
Yep, agree that a `useDebouncedCallback` hook would be useful. I moved the related changes to this PR: https://github.com/streamlit/streamlit/pull/10044